tests/http_server_test.cc

```diff
@@ -1043,3 +1051,40 @@ TEST(http_server_test, server_with_content_encoding_deflate)
     ASSERT_EQ(originalUncompressedData, newlyDecompressedData);
 }
 #endif
+
+TEST(http_server_test, http_server_is_not_leaked)
+{
+    const auto number_of_fds = [] {
+        using filesystem::directory_iterator;
+        const filesystem::path fds_dir { "/proc/self/fd" };
+
+        if (!filesystem::exists(fds_dir))
+        {
+            return directory_iterator::difference_type(0);
+        }
+
+        return std::distance(directory_iterator(fds_dir), directory_iterator {});
+    };
+
+    const auto fds_before = number_of_fds();
+    const Pistache::Address address("localhost", Pistache::Port(0));
+
+    auto server      = std::make_unique<Http::Endpoint>(address);
+    auto flags       = Tcp::Options::ReuseAddr;
+    auto server_opts = Http::Endpoint::options().flags(flags).threads(4);
+    server->init(server_opts);
+    server->setHandler(Http::make_handler<PingHandler>());
+    server->serveThreaded();
+    server->shutdown();
+    server.reset();
+
+    if (fds_before > 0)
```

Comment:
1. Under what conditions would you expect `number_of_fds()` to return 0?  gunit's `main()` does not close STDOUT nor STDERR, so there should always be at least two.
2. Why test for this condition AFTER running the HTTP server?  Why not test for it on line 1069?
3. Do you think that it is important to process an actual HTTP request within this test?
4

Reply:
1. The answer is simple - only if `/proc/self/fd` directory doesn't exist. I know that the library supports only Linux now, but I wasn't sure if I could make this assumption for unit tests. Do you recommend removing this check?
2. There is no significant reason, just to make the unit test less noisy 
3. For this particular leak - it doesn't matter. Regarding leaks in general, I think it's important to test a realistic use case. So if you recommend adding it - I will